crates/compiler/builtins/bitcode/src/str.zig

```diff
@@ -374,7 +375,12 @@ pub const RocStr = extern struct {
             return 1;
         }
 
-        const ptr: [*]usize = @as([*]usize, @ptrCast(@alignCast(self.bytes)));
+        const data_ptr = if (self.isSeamlessSlice())
```

Comment:
We check `self.isSeamlessSlice()` twice in this function. It's not an expensive operation per se, but it's multiple operations. Can we maybe avoid checking it multiple times by storing in a var or something?